src/plugins/shared_ux/public/components/page_template/no_data_page/no_data_card/types.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { EuiCardProps } from '@elastic/eui';
+import { MouseEventHandler, ReactNode } from 'react';
+
+export type NoDataPageActions = Partial<EuiCardProps> & {
+  /**
+   * Applies the `Recommended` beta badge and makes the button `fill`
+   */
+  recommended?: boolean;
+  /**
+   * Provide just a string for the button's label, or a whole component
+   */
+  button?: string | ReactNode;
```

Comment:
I'm not sure there's a specific pattern, but the idea here is that the consumer can supply just the button label (string) or replace the whole button their own (in case they don't want to use the same medium, filled button). I guess technically `string` is included in `ReactNode`, so maybe that's duplicative. But I think the explicitness here is helpful because an element would replace the whole default EuiButton where the string is just the label. Up to you all if you think there's a better pattern.

Reply:
I guess I was thinking maybe `string | EuiButtonProps`.  

```
<Card button="Click Me" />
<Card button={{ label: 'Click Me', onClick: () => {}, fill: true }} />

Then we'd have some control over precisely what component we rendered there, and could force a `fill` or `color` to keep the usage consistent, now or in the future.

It's probably minor, but I'm one who leans toward keeping the contract strict at the start, and then expanding outward as necessary, to avoid large-scale refactors later.

Reply:
You could do that. But specifically EuiButton doesn't have a `label` prop, it's label is just the `children`, so you'd likely need to append this prop type or rely on consumers to pass `children` as a key.

Reply:
Oh yeah, I think the problem with that, that I'm now remembering, is that you can't go from say an EuiButton to EuiEmptyButton by props alone since they're completely different components. So an example is if the footer button is actually not a CTA we want to promote, or say we need to provide 2 actions instead of just one, there's no way to remove the default button without replacing it completely with a `ReactNode`.